usaspending_api/disaster/v2/views/elasticsearch_base.py

```diff
@@ -219,68 +235,47 @@ def extend_elasticsearch_search_with_sub_aggregation(self, search: AwardSearch):
+        sum_covid_outlay = A("sum", field="covid_spending_by_defc.outlay", script="_value * 100")
+        sum_covid_obligation = A("sum", field="covid_spending_by_defc.obligation", script="_value * 100")
+        reverse_nested = A("reverse_nested", **{})
+        sum_loan_value = A("sum", field="total_loan_value", script="_value * 100")
+        filter_agg_query = ES_Q("terms", **{"covid_spending_by_defc.defc": self.filters.get("def_codes")})
+        filtered_aggs = A("filter", filter_agg_query)
+
+        # Apply the aggregations
+        search.aggs[self.agg_group_name].bucket(self.sub_agg_group_name, sub_group_by_sub_agg_key).bucket(
+            "nested", A("nested", path="covid_spending_by_defc")
+        ).bucket("filtered_aggs", filtered_aggs).metric("total_covid_obligation", sum_covid_obligation).metric(
+            "total_covid_outlay", sum_covid_outlay
+        ).bucket(
+            "reverse_nested", reverse_nested
+        ).metric(
+            "total_loan_value", sum_loan_value
+        )
 
     def build_totals(self, response: List[dict]) -> dict:
-        # Need to use a Postgres in this case since we only look at the first 10k results for Elasticsearch.
-        # Since the endpoint is performing aggregations on the entire matview with no grouping or joins
-        # the query takes minimal time to complete.
-        if self.agg_key == settings.ES_ROUTING_FIELD:
-            annotations = {"cast_def_codes": Cast("def_codes", ArrayField(TextField()))}
-            filters = [
-                Q(cast_def_codes__overlap=self.def_codes),
-                self.has_award_of_provided_type(should_join_awards=False),
-            ]
-            aggregations = {
-                "face_value_of_loan": Sum("total_loan_value"),
-                "obligation": Sum("obligation"),
-                "outlay": Sum("outlay"),
-            }
-            aggregations = {col: aggregations[col] for col in self.sum_column_mapping.keys()}
-            aggregations["award_count"] = Count("award_id")
-
-            if self.filters.get("query"):
-                filters.append(Q(recipient_name__icontains=self.filters["query"]["text"]))
-
-            totals = (
-                CovidFinancialAccountMatview.objects.annotate(**annotations)
-                .filter(*filters)
-                .values()
-                .aggregate(**aggregations)
-            )
-            return totals
-
         totals = {key: 0 for key in self.sum_column_mapping.keys()}
-        award_count = 0
-
-        for bucket in response:
-            for key in totals.keys():
-                totals[key] += get_summed_value_as_float(bucket, self.sum_column_mapping[key])
-            award_count += int(bucket.get("doc_count", 0))
-
-        totals["award_count"] = award_count
+        for key in totals.keys():
+            totals[key] += get_summed_value_as_float(
+                response if key != "face_value_of_loan" else response.get("reverse_nested", {}),
```

Comment:
Appears to be fine for CFDA so might just be recipient.

Reply:
I believe this is just due to the test index being out of sync with postgres - the old recipient endpoint uses postgres to get the total award counts, while the new version is using elasticsearch. If we run just a count query for awards with defc we get `18,536,702` on postgres and `16,945,918` awards on the test index. If we want to be extra sure, we can re-index the test index again and make sure it's working correctly

Reply:
Oh that is a good point. Production was just recently restored down to QAT but the ES cluster was left alone. I am fine with that then since CFDA and Agency both match up and recipient is much more than 10k.